### PR TITLE
[YW-97]Design and implement leaderboard APIs (Novels, Users, Authors).

### PR DESCRIPTION
**Pull request by LuxuryB**
🎯 Objective

Implement backend APIs to support leaderboards for Novels, Users, and Authors APIs will provide ranked lists based on different metrics (e.g., views, likes, favorites, ratings, activity) and time ranges (daily, weekly, monthly, all-time). The goal is to deliver a unified, performant, and extensible interface for the frontend to display rankings.

📦 Scope

Leaderboard Types

Novels Leaderboard

Ranked by views, votes, sort time

Users Leaderboard

Ranked by exp

Authors Leaderboard

Ranked by aggregated stats across all works (novel num, total views, total votes, sort time)

API Features

Fetch leaderboard by type (novels | users | authors)

Support metric selection (views | votes | sort time | exp)

Support time range filtering (weekly | monthly | all)

Pagination support

Return rank, entity info, and metric value

Data Source

Based on existing entities (Novel, User, Writer) and related statistics (reads, likes, favorites, comments)

Introduce aggregation queries or caching layer for performance optimization

✅ Acceptance Criteria

[ ] RESTful API endpoint: /api/leaderboards/{type}

[ ] Query parameters: metric, timeRange, page, size

[ ] Response includes: rank, entity ID, name/title, metric value

[ ] Response time ≤ 500ms (with cache hit)

[ ] Swagger/OpenAPI documentation with request/response examples

[ ] Unit tests with ≥ 80% coverage, including edge cases (empty data, invalid params, pagination boundaries)

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ spring:
     restart:
       enabled: false
   flyway:
-    enabled: true
```
Comment (phutruonnttn) [inline src/main/resources/application.yml:50]: Dont change this one, this allows local runs to use Flyway’s init files

File: src/main/java/com/yushan/backend/service/RankingService.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+     * @param categoryId
+     * @param timeRange
+     * @return
+     */
+    public PageResponseDTO<NovelDetailResponseDTO> rankNovel(
+            Integer page,
+            Integer size,
+            String sortType,
+            Integer categoryId,
+            String timeRange) {
```
Comment (phutruonnttn) [inline src/main/java/com/yushan/backend/service/RankingService.java:41]: there’s no implementation from database queries for timerange

File: src/main/java/com/yushan/backend/service/RankingService.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+     * rank user by exp
+     * @param page
+     * @param size
+     * @param timeRange
+     * @return
+     */
+    public PageResponseDTO<UserProfileResponseDTO> rankUser(
+            Integer page,
+            Integer size,
+            String timeRange) {
```
Comment (phutruonnttn) [inline src/main/java/com/yushan/backend/service/RankingService.java:72]: there’s no implementation from database queries for timerange

File: src/main/java/com/yushan/backend/service/RankingService.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+     * @param size
+     * @param sortType
+     * @param timeRange
+     * @return
+     */
+    public PageResponseDTO<AuthorResponseDTO> rankAuthor(
+            Integer page,
+            Integer size,
+            String sortType,
+            String timeRange) {
```
Comment (phutruonnttn) [inline src/main/java/com/yushan/backend/service/RankingService.java:105]: there’s no implementation from database queries for timerange

**Review by phutruonnttn — APPROVED** [review]
LGTM, please use "Squash and merge"
